soc/xtensa/intel_adsp/common/include/gdbstub/soc.h

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <inttypes.h>
+
+#ifndef SOC_XTENSA_ADSP_GDBSTUB_H_
+#define SOC_XTENSA_ADSP_GDBSTUB_H_
+
+#ifndef ZEPHYR_INCLUDE_ARCH_XTENSA_GDBSTUB_SYS_H_
+#error "Must be included after arch/xtensa/gdbstub.h"
```

Comment:
Uh... why?  The only dependency I see below is on SOC_GDB_GPKT_BIN_SIZE, which I assume is in the arch stub?  Just include that file here.